drivers/serial/uart_cc13xx_cc26xx.c

```diff
@@ -284,6 +351,119 @@ static void uart_cc13xx_cc26xx_isr(void *arg)
 
 #endif /* CONFIG_UART_INTERRUPT_DRIVEN */
 
+#ifdef CONFIG_SYS_POWER_MANAGEMENT
+/*
+ *  ======== postNotifyFxn ========
+ *  Called by Power module when waking up the CPU from Standby, to support
+ *  the case when SYS_POWER_MANAGEMENT is set but DEVICE_POWER_MANAGEMENT is
+ *  not. The uart needs to be reconfigured afterwards unless Zephyr's device
+ *  PM turned it off, in which case it'd be responsible for turning it back
+ *  on and reconfiguring it.
+ */
+static int postNotifyFxn(unsigned int eventType, uintptr_t eventArg,
+	uintptr_t clientArg)
+{
+	struct device *dev = (struct device *)clientArg;
+	int ret = Power_NOTIFYDONE;
+	s16_t res_id;
+
+	/* Reconfigure the hardware if returning from standby */
+	if (eventType == PowerCC26XX_AWAKE_STANDBY) {
+		if (get_dev_conf(dev)->regs ==
+			DT_TI_CC13XX_CC26XX_UART_40001000_BASE_ADDRESS) {
+			res_id = PowerCC26XX_PERIPH_UART0;
+		} else { /* DT_TI_CC13XX_CC26XX_UART_4000B000_BASE_ADDRESS */
+			res_id = PowerCC26X2_PERIPH_UART1;
+		}
+
+		if (Power_getDependencyCount(res_id) != 0) {
+			/*
+			 * Reconfigure and enable UART only if not
+			 * actively powered down
+			 */
+			if (uart_cc13xx_cc26xx_configure(dev,
+				&get_dev_data(dev)->uart_config) != 0) {
+				ret = Power_NOTIFYERROR;
+			}
+		}
+	}
+
+	return (ret);
+}
+#endif
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int uart_cc13xx_cc26xx_set_power_state(struct device *dev,
+	u32_t new_state)
+{
+	int ret = 0;
+
+	if ((new_state == DEVICE_PM_ACTIVE_STATE) &&
+		(new_state != get_dev_data(dev)->pm_state)) {
+		if (get_dev_conf(dev)->regs ==
+			DT_TI_CC13XX_CC26XX_UART_40001000_BASE_ADDRESS) {
+			Power_setDependency(PowerCC26XX_PERIPH_UART0);
+		} else {
+			Power_setDependency(PowerCC26X2_PERIPH_UART1);
+		}
+		/* Configure and enable UART */
+		ret = uart_cc13xx_cc26xx_configure(dev,
+			&get_dev_data(dev)->uart_config);
+		if (ret == 0) {
+			get_dev_data(dev)->pm_state = new_state;
+		}
+	} else {
+		__ASSERT_NO_MSG(new_state == DEVICE_PM_LOW_POWER_STATE ||
+			new_state == DEVICE_PM_SUSPEND_STATE ||
+			new_state == DEVICE_PM_OFF_STATE);
+
+		if (get_dev_data(dev)->pm_state == DEVICE_PM_ACTIVE_STATE) {
+			UARTDisable(get_dev_conf(dev)->regs);
+			/*
+			 * Release power dependency - i.e. potentially power
+			 * down serial domain.
+			 */
+			if (get_dev_conf(dev)->regs ==
+			    DT_TI_CC13XX_CC26XX_UART_40001000_BASE_ADDRESS) {
+				Power_releaseDependency(
+					PowerCC26XX_PERIPH_UART0);
+				get_dev_data(dev)->pm_state = new_state;
```

Comment:
Updating `pm_state` could be moved from branches to follow the conditional for clarity and to match the active state code.